sopel/bot.py

```diff
@@ -342,6 +344,37 @@ def post_setup(self) -> None:
 
     # plugins management
 
+    def store_plugin_handler(
+        self,
+        handler: AbstractPluginHandler,
+    ) -> None:
+        """Record a plugin ``handler``.
+
+        :param handler: the plugin handler to record
+
+        Recording a plugin handler associates only its name to :attr:`plugins`.
+        To register a plugin handler's callables, jobs, etc., you should use
+        its :meth:`~sopel.plugins.handlers.AbstractPluginHandler.register`
+        method.
+
+        .. versionadded:: 8.1
+        """
+        self._plugins[handler.name] = handler
+
+    def remove_plugin_handler(self, name: str) -> None:
```

Comment:
Depending on how you feel about `store_plugin_handler()` -> `set_plugin_handler()`, you might take this too:

```suggestion
    def clear_plugin_handler(self, name: str) -> None:
```